src/spatialdata/_core/operations/aggregate.py

```diff
@@ -1,148 +1,256 @@
 from __future__ import annotations
 
 import warnings
-from typing import Any
+from typing import TYPE_CHECKING, Any, Optional
 
 import anndata as ad
 import dask as da
 import dask.dataframe as ddf
 import geopandas as gpd
 import numpy as np
 import pandas as pd
+from dask.dataframe.core import DataFrame as DaskDataFrame
+from geopandas import GeoDataFrame
 from multiscale_spatial_image import MultiscaleSpatialImage
 from scipy import sparse
+from shapely import Point
 from spatial_image import SpatialImage
 from xrspatial import zonal_stats
 
 from spatialdata._core.operations.transform import transform
 from spatialdata._core.query._utils import circles_to_polygons
+from spatialdata._core.query.relational_query import get_values
 from spatialdata._types import ArrayLike
 from spatialdata.models import (
     Image2DModel,
     Labels2DModel,
     PointsModel,
     ShapesModel,
+    SpatialElement,
+    TableModel,
     get_model,
 )
 from spatialdata.transformations import BaseTransformation, Identity, get_transformation
 
+if TYPE_CHECKING:
+    from spatialdata import SpatialData
+
 __all__ = ["aggregate"]
 
 
+def _parse_element(element: str | SpatialElement, sdata: SpatialData | None, str_for_exception: str) -> SpatialElement:
+    if not ((sdata is not None and isinstance(element, str)) ^ (not isinstance(element, str))):
+        raise ValueError(
+            f"To specify the {str_for_exception!r} spatial element element, please do one of the following: "
+            f"- either pass a SpatialElement to the {str_for_exception!r} parameter (and keep "
+            f"`{str_for_exception}_sdata` = None);"
+            f"- either `{str_for_exception}_sdata` needs to be a SpatialData object, and {str_for_exception!r} needs "
+            f"to be the string name of the element."
+        )
+    if sdata is not None:
+        assert isinstance(element, str)
+        return sdata[element]
+    assert element is not None
+    return element
+
+
 def aggregate(
-    values: ddf.DataFrame | gpd.GeoDataFrame | SpatialImage | MultiscaleSpatialImage,
-    by: gpd.GeoDataFrame | SpatialImage | MultiscaleSpatialImage,
-    id_key: str | None = None,
-    *,
-    value_key: str | None = None,
-    agg_func: str | list[str] = "mean",
+    values_sdata: SpatialData | None = None,
```

Comment:
I don't understand how a function can have all default arguments, at least one should have a mandatory argument no?

Reply:
true, `values` and `by` are always mandatory, I will change the order. I made them optional because I had put them after the optional values `values_sdata` and `by_sdata`